npm/pkg/dataplane/policies/chain-management_linux.go

```diff
@@ -54,6 +53,25 @@ var (
 	ingressOrEgressPolicyChainPattern = fmt.Sprintf("'Chain %s-\\|Chain %s-'", util.IptablesAzureIngressPolicyChainPrefix, util.IptablesAzureEgressPolicyChainPrefix)
 )
 
+type osTools struct {
+	chainsToCleanup map[string]struct{}
+}
+
+func makeTools() osTools {
+	return osTools{make(map[string]struct{})}
+}
+
+func (pMgr *PolicyManager) reboot() error {
```

Comment:
would be nice to comment what we discuss. 
For example, 
```shell
A pro-active approach to avoid time to install default chains when the first networkpolicy comes again while logically only reset is called when there is no more networkpolicy and initialize is called if the first networkpolicy comes.
```